package-toolkit/runtime/src/main/kotlin/com/atlan/pkg/Utils.kt

```diff
@@ -765,6 +765,36 @@ object Utils {
             (!directUpload && objectStoreKey.isNotBlank())
     }
 
+    /**
+     * Get the list of files in the provided directory.
+     *
+     * @param importType details of the type of import expected (direct vs object store-based)
+     * @param directory path to the directory to list files from
+     * @return list of absolute paths of files in the directory
+     */
+    fun getFilesInDirectory(
+        importType: String,
+        directory: String,
+    ): List<String> {
+        if (importType == "DIRECT") {
+            val path = Paths.get(directory)
+            val files = mutableListOf<String>()
+            if (path.isDirectory()) {
+                path.walk().forEach {
+                    if (it.isRegularFile()) {
+                        files.add(it.pathString)
+                    }
+                }
```

Comment:
The file upload widget we have will only ever allow uploading a single file. So the logic here would probably need to instead look for whether the uploaded file is some kind of archive (a .zip file), and if so extract it and then walk its contents.